pkg/scheduler/admission/podnodeconstraints/admission.go

```diff
@@ -0,0 +1,201 @@
+		if err != nil {
+			return nil, err
+		}
+		return NewPodNodeConstraints(pluginConfig), nil
+	})
+}
+
+// NewPodNodeConstraints creates a new admission plugin to prevent objects that contain pod templates
+// from containing node bindings by name or selector based on role permissions.
+func NewPodNodeConstraints(config *api.PodNodeConstraintsConfig) admission.Interface {
+	return &podNodeConstraints{
+		config:  config,
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}
+}
+
+type podNodeConstraints struct {
+	*admission.Handler
+	client     client.Interface
+	config     *api.PodNodeConstraintsConfig
+	authorizer authorizer.Authorizer
+}
+
+var resourcesToAdmit = map[unversioned.GroupResource]unversioned.GroupKind{
+	kapi.Resource("pods"):                   kapi.Kind("Pod"),
+	kapi.Resource("replicationcontrollers"): kapi.Kind("ReplicationController"),
+	extensions.Resource("deployments"):      extensions.Kind("Deployment"),
+	extensions.Resource("replicasets"):      extensions.Kind("ReplicaSet"),
+	extensions.Resource("jobs"):             extensions.Kind("Job"),
+	deployapi.Resource("deploymentconfigs"): deployapi.Kind("DeploymentConfig"),
+}
+
+func shouldAdmitResource(resource unversioned.GroupResource, kind unversioned.GroupKind) (bool, error) {
+	expectedKind, shouldAdmit := resourcesToAdmit[resource]
+	if !shouldAdmit {
+		return false, nil
+	}
+	if expectedKind != kind {
+		return false, fmt.Errorf("Unexpected resource kind %v for resource %v", &kind, &resource)
+	}
+	return true, nil
+}
+
+// var _ = oadmission.WantsOpenshiftClient(&podNodeConstraints{})
+var _ = oadmission.Validator(&podNodeConstraints{})
+var _ = oadmission.WantsAuthorizer(&podNodeConstraints{})
+
+func readConfig(reader io.Reader) (*api.PodNodeConstraintsConfig, error) {
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
+		return nil, nil
+	}
+
+	obj, err := configlatest.ReadYAML(reader)
+	if err != nil {
+		return nil, err
+	}
+	if obj == nil {
+		return nil, nil
+	}
+	config, ok := obj.(*api.PodNodeConstraintsConfig)
+	if !ok {
+		return nil, fmt.Errorf("unexpected config object: %#v", obj)
+	}
+	// No validation needed since config is just list of strings
+	return config, nil
+}
+
+func (o *podNodeConstraints) Admit(attr admission.Attributes) error {
+	switch {
+	case o.config == nil,
+		attr.GetSubresource() != "":
+		return nil
+	}
+	shouldAdmit, err := shouldAdmitResource(attr.GetResource(), attr.GetKind())
+	if err != nil {
+		return err
+	}
+	if !shouldAdmit {
+		return nil
+	}
+	// Only check Create operation on pods
+	if attr.GetResource() == kapi.Resource("pods") && attr.GetOperation() != admission.Create {
+		return nil
+	}
+	ps, err := o.getPodSpec(attr)
+	if err == nil {
+		return o.admitPodSpec(attr, ps)
+	}
+	return err
+}
+
+// extract the PodSpec from the pod templates for each object we care about
+func (o *podNodeConstraints) getPodSpec(attr admission.Attributes) (kapi.PodSpec, error) {
```

Comment:
I'm still missing where you're running the through the scheme to make sure that we aren't missing resources that include pod specs.